libs/keystore/fs_keystore.go

```diff
@@ -143,3 +146,43 @@ func checkPerms(perms os.FileMode) error {
 	}
 	return nil
 }
+
+func Key(path string, keyName KeyName) ([]byte, error) {
+	expanded, err := homedir.Expand(filepath.Clean(path))
+	if err != nil {
+		return nil, err
+	}
+	ks, err := NewFSKeystore(filepath.Join(expanded, "keys"), nil)
+	if err != nil {
+		return nil, err
+	}
+
+	var key PrivKey
+	key, err = ks.Get(keyName)
+	if err != nil {
+		if !errors.Is(err, ErrNotFound) {
+			return nil, err
+		}
+		// otherwise, generate and save new priv key
+		key, err = generateNewKey(ks, keyName)
+		if err != nil {
+			return nil, err
+		}
+	}
+
+	return key.Body, nil
+}
+
+func generateNewKey(ks Keystore, keyName KeyName) (PrivKey, error) {
```

Comment:
note, this will be limited to scope of 32byte priv keys (not an issue now bc it only has one use, so we can generalise eventually)